MerlinAU.sh

```diff
@@ -806,15 +806,18 @@ _GetAllNodeSettings_()
     [ ! -d "$SETTINGS_DIR" ] && mkdir -m 755 -p "$SETTINGS_DIR"
 
     if [ -f "$SETTINGSFILE" ]; then
-        # Construct the setting key pattern to match
-        pattern="${node_mac_address}.*${setting_part}"
+        # Escape characters that might be interpreted by regex engines, like ':' in MAC addresses
+        local escaped_mac_address=$(echo "$node_mac_address" | sed 's/[\*\.]/\\&/g')
+
+        # Construct the pattern to match the entire line containing the MAC address and the setting part
+        pattern="${escaped_mac_address}.*${setting_part}=\"[^\"]*\""
 
         # Search for the setting in the settings file
-        matched_lines="$(grep -o "FW_Node.*_${setting_part}=\"[^\"]*\"" "$SETTINGSFILE" | grep -o "${pattern}=\"[^\"]*\"" || echo "")"
+        matched_lines=$(grep -o "$pattern" "$SETTINGSFILE" || echo "")
 
         if [ -n "$matched_lines" ]; then
-            # If multiple lines match, this extracts the value from the first matched line
-            setting_value="$(echo "$matched_lines" | head -n 1 | awk -F '=' '{print $2}' | tr -d '"')"
+            # Extract the value from the first matched line
```

Comment:
What's the reasoning for removing the enclosing double quotes from variable assignments?

Reply:
No specific reason I think that was just due to me copy and pasting from my old submission that worked in PR: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/179/files

![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/6145d183-f73e-42f3-8449-4a072dd5e0d0)

I normally do that incase there was some kinda change I didn't see, I can fix it I doubt this had anything to do with the fix. Was just left over from me copy and pasting from the original function when testing stuff.

Reply:
Added them back and added the new ones from my fix

Confirmed this had no impact either way, it was just left from my original method while testing.